apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -0,0 +1,111 @@
+defmodule GameBackend.CurseOfMirra.Quests do
+  @moduledoc """
+
+  """
+  alias GameBackend.Quests.DailyQuest
+  alias GameBackend.Repo
+  alias GameBackend.Quests.Quest
+  alias GameBackend.Matches.ArenaMatchResult
+  alias Ecto.Multi
+  import Ecto.Query
+
+  defmacrop custom_where_macro(queryable, binding, field_name, operator, value) do
+    {:where, [],
+     [
+       queryable,
+       [
+         {{:^, [], [binding]}, {:relation, [], Elixir}}
+       ],
+       {operator, [context: Elixir, import: Kernel],
+        [
+          {:field, [], [{:relation, [], Elixir}, {:^, [], [field_name]}]},
+          {:^, [], [value]}
+        ]}
+     ]}
+  end
+
+  def upsert_quests(quests_params) do
+    {multi, _acc} =
+      Enum.reduce(quests_params, {Multi.new(), 0}, fn quest_param, {multi, acc} ->
+        quest_changeset = Quest.changeset(%Quest{}, quest_param)
+
+        multi = Multi.insert(multi, {:insert_quest, acc}, quest_changeset)
+
+        {multi, acc + 1}
+      end)
+
+    Repo.transaction(multi)
+  end
+
+  def build_query_from_quest_description(%Quest{} = quest_description, user_id) do
```

Comment:
Need to understand a bit more why we need this and how we are going to use it, is the idea here to make 1 query per user per quest? If yes, then we are talking 30 queries to the DB (assuming 3 daily quests) per game ending. Honestly I'm preemptively worried about how this will perform under load

What do you think?

Reply:
yeah of course! i was testing different appoaches of a result , now i have something that's worth looking :P